lib/doorkeeper/oauth/authorization_code_request.rb

```diff
@@ -53,7 +53,7 @@ def validate_client
       end
 
       def validate_grant
-        return false unless grant && grant.application_id == client.id
+        return false unless grant && client && grant.application_id == client.id
```

Comment:
We have `validate :client,       error: :invalid_client`, why it passes blank client?

Reply:
In the Auth Code flow, when I was trying to exchange a code for an access token without sending the client id, it returns 500. While this is an edge case, the server should return a invalid response instead of 500 when a required params is missing.